src/core_plugins/kibana/public/dashboard/index.html

```diff
@@ -1,43 +1,57 @@
 <dashboard-app class="app-container dashboard-container">
+  <!-- Local nav. -->
   <kbn-top-nav name="dashboard" config="topNavMenu">
-    <div class="kibana-nav-info">
-      <span ng-show="dash.id" class="kibana-nav-info-title">
-        <span ng-bind="::dash.title"></span>
-      </span>
-    </div>
-  </kbn-top-nav>
-
-  <navbar ng-show="chrome.getVisible()" name="dashboard-search">
-    <form name="queryInput"
-      class="fill inline-form"
-      ng-submit="filterResults()"
-      role="form">
-
-      <div class="typeahead" kbn-typeahead="dashboard">
-        <div class="input-group"
-          ng-class="queryInput.$invalid ? 'has-error' : ''">
-
-          <input
-            type="text"
-            placeholder="Filter..."
-            aria-label="Filter input"
-            class="form-control"
-            ng-model="state.query"
-            input-focus
-            kbn-typeahead-input
-            parse-query>
-          <button type="submit" class="btn btn-default" ng-disabled="queryInput.$invalid" aria-label="Filter dashboards">
-            <span aria-hidden="true" class="fa fa-search"></span>
-          </button>
-        </div>
-
-        <kbn-typeahead-items></kbn-typeahead-items>
+    <!-- Transcluded elements. -->
+    <div data-transclude-slots>
```

Comment:
I've made a significant change to the way `kbnTopNav` works. It can now transclude content into its top left corner, and into its bottom row, using this idea of "slots", which are identified by the `data-transclude-slots` and `data-transclude-slot` attributes. Refer to the [localNav examples](https://elastic.github.io/kibana-ui-framework/#/localnav) to see how these areas of the nav are used.